src/coinbase/wallet.ts

```diff
@@ -135,30 +122,39 @@ export class Wallet {
    * @param model - The underlying Wallet model object
    * @param seed - The seed to use for the Wallet. Expects a 32-byte hexadecimal with no 0x prefix. If null or undefined, a new seed will be generated.
    * If the empty string, no seed is generated, and the Wallet will be instantiated without a seed and its corresponding private keys.
-   * @param addressModels - The models of the addresses already registered with the Wallet. If not provided, the Wallet will derive the first default address.
    * @throws {ArgumentError} If the model or client is not provided.
    * @throws {InternalError} - If address derivation or caching fails.
    * @throws {APIError} - If the request fails.
    * @returns A promise that resolves with the new Wallet object.
    */
-  public static async init(
-    model: WalletModel,
-    seed?: string | undefined,
-    addressModels: AddressModel[] = [],
-  ): Promise<Wallet> {
-    this.validateSeedAndAddressModels(seed, addressModels);
-
+  public static async init(model: WalletModel, seed?: string | undefined): Promise<Wallet> {
+    const wallet = new Wallet(model, undefined, seed);
     if (Coinbase.useServerSigner) {
-      return new Wallet(model, undefined, undefined, addressModels);
+      return wallet;
     }
-
-    const seedAndMaster = this.getSeedAndMasterKey(seed);
-    const wallet = new Wallet(model, seedAndMaster.master, seedAndMaster.seed, addressModels);
-    wallet.deriveAddresses(addressModels);
-
+    wallet.masterNode(seed);
     return wallet;
   }
 
+  /**
+   * Returns the master node for the given seed.
```

Comment:
Most of the usage of this method is to set the seed / master node. Should this be called `setMasterNode` since `setSeed` is the public method?

Reply:
Do we want this method to even return the master node?

Should this just be 

```
Sets the master node for the given seed, if valid.
If the seed is undefined it will set the master node using a random seed.
```

Reply:
makes sense, I updated to `setMasterNode`